action/editor.php

```diff
@@ -268,6 +269,18 @@ public function addAddtionalForms(Doku_Event $event)
         $datalistHTML .= '</datalist>';
         echo $datalistHTML;
     }
+
+    public function addJSINFO()
+    {
+        global $JSINFO;
+
+        $smileyConf = [];
+        $smileys = getSmileys();
+        foreach ($smileys as $syntax => $icon) {
+            $smileyConf[] = ['syntax' => $syntax, 'icon' => $icon];
+        }
+        $JSINFO['SMILEY_CONF'] = $smileyConf;
+    }
```

Comment:
Could we only do this when the WYSIWYG editor is loaded?

Reply:
But can't we toggle the editor anytime per Javascript, without re-rendering the page? In that case the data would be missing. Or do you mean something else?

Reply:
You're right. I forgot about that.